### PR TITLE
Added a converter from NRML 0.4 to NRML 0.5 for fragility functions

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-risklib/issues/435

### CODE REVIEW

File: openquake/commonlib/commands/upgrade_nrml.py [hunk 1]
```diff
@@ -83,15 +83,68 @@ def get_vulnerability_functions_04(fname):
...
+    :param fname:
+        path of the fragility file
+    :returns:
+        an :class:`openquake.commonib.node.LiteralNode` instance
+    """
+    convert_type = {"lognormal": "logncdf"}
+    new = LiteralNode('fragilityModel',
+                      dict(assetCategory='building',
+                           lossCategory='structural',
+                           id='converted-from-NRML-0.4'))
```
Comment (ptormene) [inline openquake/commonlib/commands/upgrade_nrml.py]: Probably assigning a unique value for the id would be better

File: openquake/commonlib/commands/upgrade_nrml.py [hunk 2]
```diff
@@ -83,15 +83,68 @@ def get_vulnerability_functions_04(fname):
...
+
 def upgrade_file(path):
     """Upgrade to the latest NRML version"""
     node0 = nrml.read(path, chatty=False)[0]
     shutil.copy(path, path + '.bak')  # make a backup of the original file
-    if striptag(node0.tag) == 'vulnerabilityModel':
+    tag = striptag(node0.tag)
+    if tag == 'vulnerabilityModel':
         vf_dict, cat_dict = get_vulnerability_functions_04(path)
         node0 = LiteralNode(
```
Comment (ptormene) [inline openquake/commonlib/commands/upgrade_nrml.py]: Please add a comment to state that these lines perfom the conversion from 0.4 to 0.5

File: openquake/commonlib/nrml_examples/fragm_c.xml [hunk 3]
```diff
@@ -1,60 +1,39 @@
-<?xml version='1.0' encoding='utf-8'?>
-<nrml xmlns="http://openquake.org/xmlns/nrml/0.4">
-
-    <!-- The fragility function format applies to the entire fragility model -->
-    <fragilityModel format="continuous">
-        <description>Fragility model for Pavia (continuous)</description>
-
-        <!-- limit states apply to the entire fragility model -->
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ptormene) [inline openquake/commonlib/nrml_examples/fragm_c.xml]: Please take into consideration the possibility to keep the old files in a separate folders. It might also be useful for testing the conversion.

File: openquake/commonlib/tests/riskmodels_test.py [hunk 4]
```diff
@@ -197,3 +251,33 @@ def test_ok(self):
...
         pickle.loads(pickle.dumps(cmodel))
+
+    def test_wrong_association(self):
+        scm = os.path.join(FF_DIR, 'structural_consequence_model.xml')
+        ccm = os.path.join(FF_DIR, 'contents_consequence_model.xml')
+        # exchanging the associations on purpose
+        inputs = dict(structural_consequence=ccm, contents_consequence=scm)
+        with self.assertRaises(ValueError) as ctx:
+            riskmodels.get_consequence_models(inputs)
+        self.assertIn('structural_consequence_model.xml" is associated to '
```
Comment (ptormene) [inline openquake/commonlib/tests/riskmodels_test.py]: Please fix the error message

File: openquake/commonlib/riskmodels.py [hunk 5]
```diff
@@ -160,9 +192,13 @@ def get_consequence_models(inputs):
             '(structural|nonstructural|contents|business_interruption)'
             '_consequence', key)
         if mo:
-            [node] = nrml.read(inputs[key])
-            cmodels[node['lossCategory']] = (
-                scientific.ConsequenceModel.from_node(node))
+            cmodel = nrml.parse(inputs[key])
+            if cmodel.lossCategory != mo.group(1):
+                raise ValueError(
+                    'Error in the .ini file: "%s_file=%s" is associated to a '
```
Comment (ptormene) [inline openquake/commonlib/riskmodels.py]: Please fix the error message

**Comment by ptormene** [comment]
Besides the comments, LGTM

File: openquake/commonlib/commands/upgrade_nrml.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/commands/upgrade_nrml.py]: Done

File: openquake/commonlib/commands/upgrade_nrml.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/commands/upgrade_nrml.py]: Done

File: openquake/commonlib/nrml_examples/fragm_c.xml (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/nrml_examples/fragm_c.xml]: It makes sense. Currently there is automatic test for the conversion, it will probably be added in a different pull request.

File: openquake/commonlib/tests/riskmodels_test.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/tests/riskmodels_test.py]: Done

File: openquake/commonlib/riskmodels.py (hunk 5 above)
Comment (micheles) [inline openquake/commonlib/riskmodels.py]: Done
